rflib/main/default/classes/rflib_LogArchiveController.cls

```diff
@@ -99,23 +99,34 @@ public with sharing class rflib_LogArchiveController {
     public class DefaultLogArchiveQueryLocator implements ILogArchiveQueryLocator {
 
         public List<rflib_Logs_Archive__b> getRecords(Datetime startDate, Datetime endDate) {
-            return [
-                SELECT CreatedDate__c, CreatedById__c, Context__c, Log_Level__c, Request_ID__c, Log_Messages__c, Platform_Info__c 
-                FROM rflib_Logs_Archive__b
-                WHERE CreatedDate__c > :startDate
-                AND CreatedDate__c < :endDate
-                LIMIT :ARCHIVE_LOG_QUERY_LIMIT
-            ];
+            return (List<rflib_Logs_Archive__b>) rflib_SOQL.of(rflib_Logs_Archive__b.SObjectType)
+                .with(rflib_Logs_Archive__b.CreatedDate__c)
+                .with(rflib_Logs_Archive__b.CreatedById__c)
+                .with(rflib_Logs_Archive__b.Context__c)
+                .with(rflib_Logs_Archive__b.Log_Level__c)
+                .with(rflib_Logs_Archive__b.Request_ID__c)
+                .with(rflib_Logs_Archive__b.Log_Messages__c)
+                .with(rflib_Logs_Archive__b.Platform_Info__c)
+                .whereAre(rflib_SOQL.Filter
+                    .with(rflib_Logs_Archive__b.CreatedDate__c).greaterThan(startDate)
+                    .with(rflib_Logs_Archive__b.CreatedDate__c).lessThan(endDate))
+                .setLimit(ARCHIVE_LOG_QUERY_LIMIT)
+                .mockId('DefaultLogArchiveQueryLocator.getRecords')
+                .toList();
         }
 
         public List<rflib_Logs_Archive__b> getRecordsOlderThan(Datetime referenceDate) {
-            return [
-                SELECT CreatedDate__c, Context__c, Log_Level__c, Request_ID__c 
-                FROM rflib_Logs_Archive__b 
-                WHERE CreatedDate__c < :referenceDate
-                ORDER BY CreatedDate__c ASC
-                LIMIT 50000
-            ];
+            return (List<rflib_Logs_Archive__b>) rflib_SOQL.of(rflib_Logs_Archive__b.SObjectType)
```

Comment:
Casting it's not needed 

Something like that will work as well:

```java
            return rflib_SOQL.of(rflib_Logs_Archive__b.SObjectType)
                .with(
                    rflib_Logs_Archive__b.CreatedDate__c,
                    rflib_Logs_Archive__b.Context__c,
                    rflib_Logs_Archive__b.Log_Level__c,
                    rflib_Logs_Archive__b.Request_ID__c
                )
                .orderBy(rflib_Logs_Archive__b.CreatedDate__c)
                .whereAre(rflib_SOQL.Filter
                    .with(rflib_Logs_Archive__b.CreatedDate__c).lessThan(referenceDate))
                .setLimit(ARCHIVE_LOG_QUERY_LIMIT)
                .mockId('DefaultLogArchiveQueryLocator.getOldestRecords')
                .toList();
```

Reply:
You can have up-to 5 fields in `with` method:

https://soql.beyondthecloud.dev/api/standard-soql/soql#with-field1---field5